Buddies.Web/src/pages/projects/[pid].tsx

```diff
@@ -0,0 +1,235 @@
+import Grid from '@material-ui/core/grid';
+import Container from '@material-ui/core/Container';
+import Card from '@mui/material/Card';
+import CardActionArea from '@mui/material/CardActionArea';
+import Typography from '@mui/material/Typography';
+import Avatar from '@mui/material/Avatar';
+import axios from 'axios';
+import AddIcon from '@mui/icons-material/Add';
+import EmailIcon from '@mui/icons-material/Email';
+import React, { useEffect } from 'react';
+import { useRouter } from 'next/router';
+import { authStore } from '../../stores/authStore';
+import ProjectProfilePage from '../../components/ProjectProfilePage';
+
+const baseURL = '/api/v1/projects/';
+type UserInfo = {
+  FirstName: string,
+  LastName: string,
+  UserId: number
+};
+
+type ProjectProfile = {
+  Title: string,
+  Description: string,
+  Location: string,
+  ProjectOwner: string,
+  ProjectOwnerEmail: string,
+  MaxMembers: number
+  Category: string,
+  MemberLst: UserInfo[],
+  InvitedLst: UserInfo[]
+};
+
+const memberLst: UserInfo[] = [{
+  FirstName: 'John',
+  LastName: 'Doe',
+  UserId: -1,
+}];
+
+// default project that loads when project id is not found
+const defaultProject: ProjectProfile = {
+  Title: 'Not Found',
+  Description: 'This error means the project that you are searching for was not found. There are many causes for this. First, check if the url is correct. Second, if the url is correct, it means the project owner has most likely deleted his account!',
+  Location: 'Unknown',
+  ProjectOwner: 'D.N.E',
+  ProjectOwnerEmail: 'D.N.E',
+  MaxMembers: 1,
+  Category: 'Not Found',
+  MemberLst: memberLst,
+  InvitedLst: [],
+};
+
+let projectId: string | string[] | undefined = '';
+let currId: number; // id of the user viewing the page
+const invitedIds: number[] = []; // list of userId's in the invited list
+const memberIds: number[] = []; // list of userId's in the member list
+let ownerId: number; // Id of the owner of the project
+
+/* Project profile page. Responsible for putting all the components that make up the
+  page together. It also sends GET requests to get a project by its id. And a
+  PUT request, to let invited users join the project.
+*/
+const Project: React.VFC = () => {
+  const [project, setProject] = React.useState<ProjectProfile>(defaultProject);
+  const authState = authStore((state: { authState: any; }) => state.authState);
+  const router = useRouter();
+
+  /* Gets project by id and then creates necessary global data structures.
+     (memberLst, invitedLst, project object)
+  */
+  function getProject() {
+    if (!(typeof projectId === 'string')) {
+      alert('error');
+      return;
+    }
+    axios.get(baseURL + projectId).then((res) => {
```

Comment:
Move the API calling logic to `api/index.ts` for consistency and reusability.

Reply:
Fixed